nmostesting/GenericTest.py

```diff
@@ -439,7 +451,10 @@ def do_test_api_resource(self, resource, response_code, api):
         else:
             return None
 
-        valid, response = self.do_request(resource[1]['method'], url)
+        headers = None
+        if resource[1]['method'].upper() == "OPTIONS":
+            headers = self.prepare_CORS("OPTIONS")
```

Comment:
I think this needs to generate Access-Control-Request-Headers from `self.apis[api]["spec"].get_methods(resource[0])` as below, right, whereas it looks like right now it would only add OPTIONS?